### DISCUSSION TITLE
Binary serialization

### DISCUSSION THREAD

**Discussion by pmp-gh** [NVIDIA-Omniverse/PhysX#141] [Q&A] [answered]
Hi, I would like to make some minor changes (add instance variables) to some PhysX vehicle classes for an experiment. I see a comment in the header file about layout changes breaking binary serialization. What do I have to do to not break serialization after modifying the classes? Thank you!

**Comment by preist-nvidia** [comment, marked as answer]
The binary serialization comments refer to compatibility of binary serialized data that breaks if class data layouts change. If we change any layout between PhysX versions, we change the binary data version, see `PX_BINARY_SERIAL_VERSION`.

So if you are the only dev using the changes (which is likely if you are doing an experiment) you don't have to worry about it.

**Reply by pmp-gh** [reply]
Thanks!
